### ISSUE TITLE
Implement Web Share API in share timetable modal 

### ISSUE THREAD

**Issue by ZhangYiJiang** [nusmodifications/nusmods#1290]
[Chrome Mobile 61 implements a new way to access native sharing dialog via the Web Share API](https://developers.google.com/web/updates/2016/09/navigator-share). This API is not well supported outside of Chrome, but [Caniuse indicates the next version of Safari may also implement this](https://caniuse.com/#search=web%20share). Since the API looks relatively straightforward I think we can try implementing this. 

The change should be made to the share modal. The headings will probably need to be changed, although I'm not entirely sure what would be good - maybe replace both columns on the right with a single share button if native share API is detected? 

![screenshot from 2018-11-14 10-16-07](https://user-images.githubusercontent.com/445650/48455485-4fb47e00-e7f6-11e8-805a-1e23f3447297.png)

**Comment by aaditkamat** [comment]
I would like to try working on this issue.

**Comment by ZhangYiJiang** [comment]
@aaditkamat Yeah, absolutely! Sorry I didn't notice this earlier. You don't need our permission to start on any of these issues :)

**Comment by shimeilee** [comment]
@aaditkamat Hi are you still working on it? :)

**Comment by aaditkamat** [comment]
Hi @shimeilee, I'm not currently working on this. You can take up the issue if you like.

**Comment by ravern** [comment]
Hey, we're doing some clean ups over the summer. Closing this for now because this issue is stale. Feel free to re-open this if you think otherwise!
